packages/gatsby/src/query/graphql-runner.ts

```diff
@@ -164,9 +165,20 @@ export default class GraphQLRunner {
       if (typeof statsQuery !== `string`) {
         statsQuery = statsQuery.body
       }
-      this.stats.uniqueOperations.add(`${statsQuery}${JSON.stringify(context)}`)
-
-      this.stats.uniqueQueries.add(statsQuery)
+      this.stats.uniqueOperations.add(
+        crypto
+          .createHash(`sha1`)
+          .update(statsQuery)
+          .update(JSON.stringify(context))
```

Comment:
We might not need the JSON.stringify anymore since

Reply:
This would also break on circular references (for the sake of creating a hash).

Can we either apply Sid's suggestion or at least use `v8.serialize()` for this? (That does support circular references, and a larger payload, although I certainly hope that's never an issue in this step)